homeassistant/components/jewish_calendar/__init__.py

```diff
@@ -1 +1,119 @@
 """The jewish_calendar component."""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.const import CONF_LATITUDE, CONF_LONGITUDE, CONF_NAME
+from homeassistant.helpers.discovery import async_load_platform
+import homeassistant.helpers.config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = "jewish_calendar"
+
+SENSOR_TYPES = {
+    "binary": {
+        "issur_melacha_in_effect": ["Issur Melacha in Effect", "mdi:power-plug-off"]
+    },
+    "data": {
+        "date": ["Date", "mdi:judaism"],
+        "weekly_portion": ["Parshat Hashavua", "mdi:book-open-variant"],
+        "holiday_name": ["Holiday name", "mdi:calendar-star"],
+        "holiday_type": ["Holiday type", "mdi:counter"],
+        "omer_count": ["Day of the Omer", "mdi:counter"],
+    },
+    "time": {
+        "first_light": ["Alot Hashachar", "mdi:weather-sunset-up"],
+        "gra_end_shma": ['Latest time for Shm"a GR"A', "mdi:calendar-clock"],
+        "mga_end_shma": ['Latest time for Shm"a MG"A', "mdi:calendar-clock"],
+        "plag_mincha": ["Plag Hamincha", "mdi:weather-sunset-down"],
+        "first_stars": ["T'set Hakochavim", "mdi:weather-night"],
+        "upcoming_shabbat_candle_lighting": [
+            "Upcoming Shabbat Candle Lighting",
+            "mdi:candle",
+        ],
+        "upcoming_shabbat_havdalah": ["Upcoming Shabbat Havdalah", "mdi:weather-night"],
+        "upcoming_candle_lighting": ["Upcoming Candle Lighting", "mdi:candle"],
+        "upcoming_havdalah": ["Upcoming Havdalah", "mdi:weather-night"],
+    },
+}
+
+CONF_DIASPORA = "diaspora"
+CONF_LANGUAGE = "language"
+CONF_CANDLE_LIGHT_MINUTES = "candle_lighting_minutes_before_sunset"
+CONF_HAVDALAH_OFFSET_MINUTES = "havdalah_minutes_after_sunset"
+
+CANDLE_LIGHT_DEFAULT = 18
+
+DEFAULT_NAME = "Jewish Calendar"
+
+CONFIG_SCHEMA = vol.Schema(
+    {
+        DOMAIN: vol.Schema(
+            {
+                vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+                vol.Optional(CONF_DIASPORA, default=False): cv.boolean,
+                vol.Optional(CONF_LATITUDE): cv.latitude,
+                vol.Optional(CONF_LONGITUDE): cv.longitude,
+                vol.Optional(CONF_LANGUAGE, default="english"): vol.In(
+                    ["hebrew", "english"]
+                ),
+                vol.Optional(
+                    CONF_CANDLE_LIGHT_MINUTES, default=CANDLE_LIGHT_DEFAULT
+                ): int,
+                # Default of 0 means use 8.5 degrees / 'three_stars' time.
+                vol.Optional(CONF_HAVDALAH_OFFSET_MINUTES, default=0): int,
+            }
+        )
+    },
+    extra=vol.ALLOW_EXTRA,
+)
+
+
+async def async_setup(hass, config):
+    """Set up the Jewish Calendar component."""
+    import hdate
+
+    _LOGGER.debug("Configuration loaded: %s", config)
+
+    name = config[DOMAIN].get(CONF_NAME)
```

Comment:
Use `dict[key]` for required config keys and keys with default config schema values.

Reply:
No keys are required

Reply:
If there are default values, please also use `dict[key]`. We should only use `dict.get` if we don't know that the key is in the dict.